graphql/resolvers.go

```diff
@@ -288,6 +290,83 @@ func (r *mutationResolver) SpawnHost(ctx context.Context, spawnHostInput *SpawnH
 	return &apiHost, nil
 }
 
+func (r *mutationResolver) EditSpawnHost(ctx context.Context, editSpawnHostInput *EditSpawnHostInput) (*restModel.APIHost, error) {
+	usr := route.MustHaveUser(ctx)
+	h, err := host.FindOneByIdOrTag(editSpawnHostInput.HostID)
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error finding host by id: %s", err))
+	}
+
+	if !CanUpdateSpawnHost(h, usr) {
+		return nil, Forbidden.Send(ctx, "You are not authorized to modify this host")
+	}
+
+	if editSpawnHostInput.DisplayName != nil {
+		err = h.SetDisplayName(*editSpawnHostInput.DisplayName)
+		if err != nil {
+			return nil, InternalServerError.Send(ctx, fmt.Sprintf("Error while modifying spawnhost name: %s", err))
+		}
+	}
+	if editSpawnHostInput.NoExpiration != nil {
+		if *editSpawnHostInput.NoExpiration {
```

Comment:
can this code block be replaced by the logic in this function? https://github.com/evergreen-ci/evergreen/blob/master/cloud/ec2.go#L718
if so, we add this option to a parameter in the `cloud.ModifySpawnHost` fn call and remove `ExpireInDays` from `graphql/util`